src/apps/vpn/controller.cpp

```diff
@@ -128,6 +128,17 @@ void Controller::initialize() {
     logger.info() << "Canary Ping Succeeded";
   });
 
+  connect(SettingsHolder::instance(), &SettingsHolder::transactionBegan, this,
+          [this]() { m_connectedBeforeTransaction = m_state == StateOn; });
+
+  connect(SettingsHolder::instance(),
+          &SettingsHolder::transactionAboutToRollBack, this, [this]() {
```

Comment:
I don't have enough background yet on the settings transaction setup. Why do we activate before rollback rather than after rollback? Based on nothing but naming I have thought we want to activate after rollback when our selected server is restored to it's original value. I'm sure this is just me not understanding I'm asking so I can get up to sped.

Reply:
We want it here because, in case of a disconnect, if we do it after, first we end up doing a server switch, then a disconnect.

Reply:
Matt and i are working on it. I can review this PR.